From 8a291f61cac2988112a84fcb37b771b9e291eedd Mon Sep 17 00:00:00 2001
From: Christophe Marin <christophe@krop.fr>
Date: Thu, 20 Mar 2025 13:55:40 +0100
Subject: [PATCH] Make kdbg compatible with CMake 4

Required CMake versions older than 3.5 will throw errors.

Bump the minimum version to 3.16
---
 CMakeLists.txt | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5f93b44..8108af0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,9 +1,11 @@
-cmake_minimum_required(VERSION 3.3)
-project(KDbg)
+cmake_minimum_required(VERSION 3.16)
+
+set(KDBG_VERSION 3.1.0)
+project(KDbg VERSION ${KDBG_VERSION})
+
 set(QT_MIN_VERSION "5.5.0")
 set(KF5_MIN_VERSION "5.2.0")
 
-set(KDBG_VERSION 3.1.0)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/kdbg/version.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/kdbg/version.h)
 
 find_package(ECM REQUIRED NO_MODULE)
-- 
2.49.0

